flexmeasures/api/v3_0/assets.py

```diff
@@ -506,6 +502,9 @@ def patch(self, asset_data: dict, id: int, db_asset: GenericAsset):
                             f"Updated Attr: {attr_key}, From: {current_attributes.get(attr_key)}, To: {attr_value}"
                         )
                 continue
+            if k == "flex_context":
+                # Validate the flex context schema
+                DBFlexContextSchema().load(v)
```

Comment:
We should not return a 500 error when this does not validate, but actually a 422 error.